internal/app_config.go

```diff
@@ -126,14 +132,18 @@ func (c AppConfig) findWritableFile(keyVals map[string]string) (*cfgFile, error)
 	for i := range c.cfgs {
 		for k := range keyVals {
 			if c.cfgs[i].tree.Has(k) {
-				// We also have to assert the file we are writing isn't the z-50-fioctl.toml
-				// file. This file is "managed" so doing a config write to a keyval in that
-				// file would never work - it would get overwritten.
-				if c.cfgs[i].name == "z-50-fioctl.toml" {
+				if isConfigManaged(c.cfgs[i].name) {
 					return nil, fmt.Errorf("unable to override config-managed file: %s", c.cfgs[i].path)
 				}
 
 				if !isWritable(c.cfgs[i].path) {
+					// Work our way back up more significant files to see if there
+					// is one we can update this value from
+					for j := i + 1; j < len(c.cfgs); j++ {
```

Comment:
It looks like we have to decrease `j` starting from `i-1` here?

Reply:
https://github.com/foundriesio/fioconfig/pull/45/commits/1c144006b76688346a22d73de284202739de9bf8